IntegrationTests/UI.IntegrationTests/UserControls/RevisionGrid/ToggleBetweenArtificialAndHeadCommitsTests.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Windows.Forms;
+using CommonTestUtils;
+using FluentAssertions;
+using GitCommands;
+using GitUI;
+using GitUI.CommandsDialogs;
+using GitUIPluginInterfaces;
+using NUnit.Framework;
+
+namespace GitExtensions.UITests.UserControls.RevisionGrid
+{
+    [Apartment(ApartmentState.STA)]
+    public class ToggleBetweenArtificialAndHeadCommitsTests
+    {
+        // Created once for the fixture
+        private ReferenceRepository _referenceRepository;
+        private string _initialCommit;
+        private string _headCommit;
+
+        // Created once for each test
+        private GitUICommands _commands;
+
+        [OneTimeSetUp]
+        public void SetUpFixture()
+        {
+            // There is no need to restore the original AppSettings because AppSettings is routed to a temp folder.
+
+            // We don't want avatars during tests, otherwise we will be attempting to download them from gravatar.
+            AppSettings.ShowAuthorAvatarColumn = false;
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            _referenceRepository = new ReferenceRepository();
+            _initialCommit = _referenceRepository.CommitHash;
+            _referenceRepository.CreateCommit("head commit");
+            _headCommit = _referenceRepository.CommitHash;
+
+            _commands = new GitUICommands(_referenceRepository.Module);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _referenceRepository.Dispose();
+        }
+
+        [Test]
+        public void ToggleBetweenArtificialAndHeadCommits_no_empty([Values(false, true)] bool showGitStatusForArtificialCommits)
+        {
+            RunToggleBetweenArtificialAndHeadCommitsTest(
+                showGitStatusForArtificialCommits,
+                revisionGridControl =>
+                {
+                    var hasChangesWorkTree = revisionGridControl.GetChangeCount(ObjectId.WorkTreeId).HasChanges;
+                    var hasChangesIndex = revisionGridControl.GetChangeCount(ObjectId.IndexId).HasChanges;
+                    hasChangesWorkTree.Should().Be(showGitStatusForArtificialCommits);
+                    hasChangesIndex.Should().Be(showGitStatusForArtificialCommits);
+
+                    revisionGridControl.GoToRef(_initialCommit, showNoRevisionMsg: false);
+                    revisionGridControl.LatestSelectedRevision.Guid.Should().Be(_initialCommit);
+
+                    revisionGridControl.ToggleBetweenArtificialAndHeadCommits();
+                    revisionGridControl.LatestSelectedRevision.Guid.Should().Be(GitRevision.WorkTreeGuid);
```

Comment:
How do this work when showGitStatusForArtificialCommits=false?
Add comment?

Reply:
assume has changes always, see below